### ISSUE TITLE
VCS: "Save As" of a tracked file causes traceback

### ISSUE THREAD

**Issue by uliska** [wbsoft/frescobaldi#1016]
This is a follow-up issue to #1010 whose *original* problem has been fixed by now.

When saving a tracked file with another name the file is saved but causes this traceback:

```
Traceback (most recent call last):
  File "/home/urs/git/lilypond/frescobaldi/frescobaldi_app/sidebar/__init__.py", line 228, in updateView
    self._vcsdiffarea.setTextEdit(view)
  File "/home/urs/git/lilypond/frescobaldi/frescobaldi_app/widgets/vcsdiffarea.py", line 37, in setTextEdit
    self._textedit.vcsDocTracker.diff_updated.disconnect(self.update)
TypeError: disconnect() failed between 'diff_updated' and 'update'
```

Changing the document results in this traceback:

```
Traceback (most recent call last):
  File "/home/urs/git/lilypond/frescobaldi/frescobaldi_app/vcs/gitdoc.py", line 388, in set_status
    self.status_updated.emit(self)
RuntimeError: wrapped C/C++ object of type Document has been deleted
```

When closing that new document the following traceback is shown:

```
Traceback (most recent call last):
  File "/home/urs/git/lilypond/frescobaldi/frescobaldi_app/tabbar.py", line 125, in slotTabCloseRequested
    self.window().closeDocument(self.docs[index])
  File "/home/urs/git/lilypond/frescobaldi/frescobaldi_app/mainwindow.py", line 589, in closeDocument
    doc.close()
  File "/home/urs/git/lilypond/frescobaldi/frescobaldi_app/document.py", line 114, in close
    app.documentClosed(self)
  File "/home/urs/git/lilypond/frescobaldi/frescobaldi_app/signals.py", line 191, in emit
    l.call(args, kwargs)
  File "/home/urs/git/lilypond/frescobaldi/frescobaldi_app/signals.py", line 308, in call
    return self.func(obj, *args[self.argslice], **kwargs)
  File "/home/urs/git/lilypond/frescobaldi/frescobaldi_app/viewmanager.py", line 342, in slotDocumentClosed
    space.removeDocument(doc)
  File "/home/urs/git/lilypond/frescobaldi/frescobaldi_app/viewmanager.py", line 169, in removeDocument
    self.disconnectView(self.activeView())
  File "/home/urs/git/lilypond/frescobaldi/frescobaldi_app/viewmanager.py", line 197, in disconnectView
    self.disconnectVcsLabels(view)
  File "/home/urs/git/lilypond/frescobaldi/frescobaldi_app/viewmanager.py", line 207, in disconnectVcsLabels
    view.vcsDocTracker.status_updated.disconnect(self.updateVcsDocStatus)
TypeError: 'method' object is not connected
```
* Interestingly this does *not* happen when the file is saved outside of a Git repository.
* Also interestingly it *does* happen when the file is saved as/replaces another tracked file

My impression is that we haven't properly mirrored the difference between creating a new View and changing a document URL within a view to the VCS code base.

**Comment by uliska** [comment]
I would strongly suggest to fix #1015 first because it may actually be the cause for this too.

**Comment by uliska** [comment]
As expected, fixing #1015 made this go away too :-)
